contrib_bots/lib/spotify.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+import json
+import urllib
+class SpotifyHandler(object):
+    '''
+    This plugin will allow users to directly search
+    for song, artists, and albums using the Spotify API.
+    It can be called using `@spotify song/artist/album`
+    '''
+    def usage(self):
+        return '''
+            This plugin will allow users to directly search
+            for song, artists, and albums using the Spotify API
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+        original_content = message['content']
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'spotify':
+            return False
+        is_follow_up = original_content.startswith('@spotify')
+
+        return is_follow_up
+
+    def artist_content(artist):
+        artist_name = artist['name']
+        artist_url = artist['external_urls']['spotify']
+        return (artist_name, artist_url)
+    def album_content(album):
+        album_name = album['name']
+        album_url = album['external_urls']['spotify']
+        artist_name = album['artists'][0]['name']
+        artist_url = album['artists'][0]['external_urls']['spotify']
+        return (album_name, album_url, artist_name, artist_url)
+    def track_content(track):
+        song_name = track['name']
+        song_url = track['external_urls']['spotify']
+        album_name = track['album']['name']
+        album_url = track['album']['external_urls']['spotify']
+        artist_name = track['artists'][0]['name']
+        artist_url = track['artists'][0]['external_urls']['spotify']
+        return (song_name, song_url, album_name, album_url, artist_name, artist_url)
+
+    def handle_message(self, message, client, state_handler):
+        base_url = 'https://api.spotify.com/v1/search?q='
+        original_content = message['content']
+        original_content = original_content.replace('@spotify ', '')
+        new_content = ''
+        if original_content.split(' ', 1)[0] == "album":
+            new_content += " Here are the albums with \"%s\" in them: " % (original_content[6:])
+            encoded_message = urllib.parse.quote(original_content[6:])
+            url = base_url+encoded_message+"&type=album"+"&limit=10"
+            raw_data = urllib.request.urlopen(url).read()
+            raw_albums_list = json.loads(raw_data)['albums']['items']
+            for album in raw_albums_list:
+                new_content += "\n [%s](%s) by [%s](%s)\n" % album_content(album)
+
+        if original_content.split(' ')[0] == "artist":
```

Comment:
May be you want to use `elif`. It would be great if you could use a function or loop to reduce code repetition.